src/wiktextract/extractor/en/example.py

```diff
@@ -0,0 +1,237 @@
+    template_node: TemplateNode,
+    sense_data: SenseData,
+    parent_example: ExampleData,
+) -> list[ExampleData]:
+    # https://en.wiktionary.org/wiki/Template:zh-x
+    expanded_node = wxr.wtp.parse(
+        wxr.wtp.node_to_wikitext(template_node), expand_all=True
+    )
+    clean_node(wxr, sense_data, expanded_node)
+    has_dl_tag = False
+    results = []
+    for dl_tag in expanded_node.find_html_recursively("dl"):
+        has_dl_tag = True
+        ref = ""
+        roman = ""
+        translation = clean_node(
+            wxr, None, template_node.template_parameters.get(2, "")
+        )
+        roman_raw_tags = []
+        for dd_tag in dl_tag.find_html("dd"):
+            dd_text = clean_node(wxr, None, dd_tag)
+            if dd_text.startswith("From:"):
+                ref = dd_text.removeprefix("From:")
+            else:
+                for span_tag in dd_tag.find_html_recursively(
+                    "span", attr_name="lang", attr_value="Latn"
+                ):
+                    roman = clean_node(wxr, None, span_tag)
+                    for span_tag in dd_tag.find_html_recursively("span"):
+                        span_text = clean_node(wxr, None, span_tag)
+                        if span_text.startswith("[") and span_text.endswith(
+                            "]"
+                        ):
+                            roman_raw_tags.append(span_text.strip("[]"))
+
+        example_text = ""
+        last_span_is_exmaple = False
+        for span_tag in dl_tag.find_html_recursively("span"):
+            if span_tag.attrs.get("class", "") in ["Hant", "Hans"]:
+                example_text = clean_node(wxr, None, span_tag)
+                last_span_is_exmaple = True
+            elif last_span_is_exmaple:
+                last_span_is_exmaple = False
+                if len(example_text) > 0:
+                    example = deepcopy(parent_example)
+                    # dialect and character variant tag
+                    for link_node in span_tag.find_child_recursively(
+                        NodeKind.LINK
+                    ):
+                        example["raw_tags"].append(
+                            clean_node(wxr, None, link_node)
+                        )
+                    example["text"] = example_text
+                    example["roman"] = roman
+                    example["english"] = translation
+                    example["raw_tags"].extend(roman_raw_tags)
+                    if len(ref) > 0:  # don't override parent quote-* template
+                        example["ref"] = ref
+                    clean_example_empty_data(example)
+                    results.append(example)
+
+    # no source, single line example
+    if not has_dl_tag:
+        roman = ""
+        raw_tags = []
+        for span_tag in expanded_node.find_html(
+            "span", attr_name="lang", attr_value="Latn"
+        ):
+            roman = clean_node(wxr, None, span_tag)
```

Comment:
Maybe add `break` or a comment explaining why this for loop "overwrites" `roman`; I know there's only one `span` coming from the generator, but it looks a bit strange at first sight.